pkg/kn/commands/service/configuration_edit_flags.go

```diff
@@ -102,6 +104,13 @@ func (p *ConfigurationEditFlags) Apply(service *servingv1alpha1.Service, cmd *co
 		return err
 	}
 
+	if cmd.Flags().Changed("port") {
```

Comment:
i think it's a bit weird for us to not set container port _if_ we are going to specify it as a default in the above flag definition.

Reply:
Right! `Apply` gets called for `create` as well `update` both.

@savitaashture: We'll probably need to identify the command in `Apply` definition. 
`cmd.Name()` returns the command name, we can update this check to not set the port when `cmd.Name() == "update" && !cmd.Flags().Changed("port") ` and set for all other cases.

WDYT ?

Reply:
@navidshaikh yes current logic will be called while ```create``` and ```update```

Reply:
@navidshaikh there is one more scenario
consider 
1. I have created a service ```test``` with ```imagename helloworld which runs on 8888```
2. want to update the same service ```test``` with ```imagename httpproxy which runs on 9999```

then in 2nd case update will be successful for image but port will not be updated.

So we need to consider when there is update in image and if there is change in port then also need to execute logic in ```update``` scenario

Reply:
@cppforlife: 
> i think it's a bit weird for us to not set container port if we are going to specify it as a default in the above flag definition.

Do you mean, we should update the port value to default one for `service update`, even if user doesnt specify `-p` port for `service update` command ?

Reply:
@savitaashture : or we just remove the default value of port and the mentioned conditional should suffice.

Reply:
yes we can just remove default port value 8080 from the flag and ```if cmd.Flags().Changed("port") {``` condition will work for all the scenario.

Reply:
i think it's best to rely on the server default and not specify default in the cli (shows up in -h). if user does specify the port configuration then it should be applied.

Reply:
yes i have removed saying default value from (shows up in -h) and if user specify it will be applied.